packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -43,7 +43,7 @@ class CurlyComponentManager {
   create(definition, args, dynamicScope, hasBlock) {
     let parentView = dynamicScope.view;
 
-    let klass = definition.ComponentClass;
+    let klass = definition.getComponentClass(args);
```

Comment:
@krisselden are you talking about using `DynamicComponentSyntax`? If so, I'll need to modify it to support using named params, not just assuming first positional.